pkg/live/inventoryrg.go

```diff
@@ -249,78 +298,624 @@ func (icm *InventoryResourceGroup) GetObject() (*unstructured.Unstructured, erro
 	return invCopy, nil
 }
 
+// sortResourceGroups is a utility function for sorting a list of sharded ResourceGroup objects by the
+// ResourceGroup object name.
+func sortResourceGroups(resourcegroups []unstructured.Unstructured) {
+	sort.Slice(resourcegroups, func(i, j int) bool {
+		return resourcegroups[i].GetName() < resourcegroups[j].GetName()
+	})
+}
+
 // Apply is a Storage interface function implemented to apply the inventory
 // object.
 func (icm *InventoryResourceGroup) Apply(dc dynamic.Interface, mapper meta.RESTMapper, statusPolicy inventory.StatusPolicy) error {
```

Comment:
This Apply function now has a LOT more code in it. Can you add some unit tests to make sure it does what you expect?

Reply:
Same for ApplyWithPrune.

Reply:
Added tests for both funcs.